scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -3114,35 +3118,56 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) {
         name match {
           case q: Quasi if endTparamQuasi => q.become[Type.Param]
           case _ =>
-            val tparams = typeParamClauseOpt(ownerIsType = true, ctxBoundsAllowed = false)
+            val tparams = typeParamClauseOpt(
+              ownerIsType = true,
+              ctxBoundsAllowed = dialect.allowImprovedBoundsAndGivens
+            )
             val tbounds = typeBounds()
             val vbounds = new ListBuffer[Type]
             val cbounds = new ListBuffer[Type]
             if (ctxBoundsAllowed) {
               @inline
-              def getBound(): Type = {
-                next()
-                currToken match {
-                  case t: Ellipsis => ellipsis[Type](t, 1)
-                  case _ => typ()
```

Comment:
side comment, mostly rhetorical: i was surprised to find that `typ()` doesn't handle `Ellipsis`... why wouldn't it?

Reply:
We can do a follow up removing it and check, but I wouldn't do it here